src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,177 @@
+     *
+     * @param input The user input string.
+     * @param list The TaskList containing all tasks.
+     * @return The Task to be deleted.
+     */
+    public Task getTaskToDelete(String input, TaskList list) {
+        int indexOfTaskToDelete = Integer.parseInt(input.substring(7)) - 1;
+        return list.getTask(indexOfTaskToDelete);
+    }
+    
+    /**
+     * Parses the input command to retrieve the keyword the user is finding in the task list.
+     *
+     * @param input The user input string.
+     * @return The String that is in the descriptions of the tasks in task list.
+     */
+    public String getKeywordForFind(String input) {
+        return input.substring(5);
+    }
+    
+    /**
+     * Creates a ToDo task from the input command.
+     *
+     * @param input The user input string.
+     * @return A new ToDo Task.
+     */
+    public Task createToDo(String input) {
+        String todoDescription = input.substring(5);
+        return new ToDo(todoDescription);
+    }
+    
+    /**
+     * Creates a Deadline task from the input command, including parsing the due date.
+     *
+     * @param input The user input string.
+     * @param ui The UI instance to handle user interactions.
+     * @return A new Deadline Task.
+     * @throws DukeException If the input date is invalid or the input format is incorrect.
+     */
+    public Task createDeadline(String input, Ui ui) throws DukeException{
+        int indexOfBy = input.indexOf("/by");
+        String deadlineDescription = input.substring(9, indexOfBy - 1);
+        String deadline = input.substring(indexOfBy + 4);
+        LocalDate by = this.formatInputDate(deadline);
+        ui.handleInvalidInputDate(deadline, DateTimeFormatter.ofPattern("dd/MM/yyyy"), by);
+        return new Deadline(deadlineDescription, by);
+    }
+    
+    /**
+     * Creates an Event task from the input command, including parsing the start and end dates.
+     *
+     * @param input The user input string.
+     * @param ui The UI instance to handle user interactions.
+     * @return A new Event Task.
+     * @throws DukeException If any input date is invalid or the input format is incorrect.
+     */
+    public Task createEvent(String input, Ui ui) throws DukeException{
+        int indexOfFrom = input.indexOf("/from");
+        int indexOfTo = input.indexOf("/to");
+        String eventDescription = input.substring(6, indexOfFrom - 1);
+        String startString = input.substring(indexOfFrom + 6, indexOfTo - 1);
+        String endString = input.substring(indexOfTo + 4);
+        LocalDate start = this.formatInputDate(startString);
+        LocalDate end = this.formatInputDate(endString);
+        ui.handleInvalidInputDate(startString, DateTimeFormatter.ofPattern("dd/MM/yyyy"), start);
+        ui.handleInvalidInputDate(endString, DateTimeFormatter.ofPattern("dd/MM/yyyy"), end);
+        return new Event(eventDescription, start, end);
+    }
```

Comment:
Maybe can separate this chunk of code with a few blank lines to enhance readability?